### DISCUSSION TITLE
Including RequireJS

### DISCUSSION THREAD

**Discussion by a-r-j** [pradyunsg/furo#390] [Potential Issue] [answered]
Hiya, thanks for the fantastic theme!

I'm trying to populate my docs with some notebook tutorial that make use of interactive plotly plots. I use `nblink` and `nbsphinx`. I've narrowed down the problem to `furo` - the notebooks render correctly with the `sphinx-book-theme` and `alabaster` but not with `furo`.

When loading the pages with the plots, the console indicates that `requirejs` can't be found:

```
Uncaught ReferenceError: require is not defined
```

despite including both: 

```python
# conf.py
nbsphinx_require_js_path = (
    "https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"
)
...
def setup(app):
    app.add_js_file(
        "https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"
    )
```

in `conf.py`.

I have also tried with:

```python
# conf.py
html_js_files = [
    'js/require.js',
]
```

using a local copy of https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js in `js/require.js`

EDIT:

I managed to resolve my problem by setting:

```python
# conf.py
os.environ["PLOTLY_RENDERER"] = "sphinx_gallery"
```
which seems to bundle the required JS

**Comment by pvcraven** [comment, marked as answer]
Have you tried:
```py
# conf.py
html_js_files = [
    'https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js',
]

**Reply by a-r-j** [reply]
Yep, in the end I had to go with specifying the `”sphinx_gallery` Renderer for plotly
